Function/BookFunc/GenerateMeetingFunction.cs

```diff
@@ -12,63 +12,50 @@
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 using Microsoft.Graph;
-using Microsoft.Identity.Client;
-using Newtonsoft.Json;
-using Microsoft.Graph.Auth;
-using System.Web.Http;
-using System.Net;
-using TeamsMeetingBookFunc;
+using TeamsMeetingBookFunc.Models;
+using TeamsMeetingBookFunc.Services;
+using TeamsMeetingBookFunc.Helpers;
 
 namespace TeamsMeetingBookingFunction
 {
     public static class GenerateMeetingFunction
     {
-        #region config keys
-        private const string TenantIdCfg = "TenantID";
-        private const string ClientIdCfg = "ClientID";
-        private const string UserPasswordCfg = "UserPassword";
-        private const string UserEmailCfg = "UserEmail";
-        private const string DefaultMeetingNameCfg = "DefaultMeetingName";
-        #endregion
-
         [FunctionName("GenerateMeetingFunction")]
         public static async Task<IActionResult> Run(
-            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)]
+            RequestModel requestModel,
+            // HttpRequest is still passed on but currently not used
+            HttpRequest _,
             ILogger log, ExecutionContext context)
         {
             var config = BuildConfig(context);
 
-            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
-            dynamic data = JsonConvert.DeserializeObject(requestBody);
-            DateTime startDate = data?.startDateTime ?? DateTime.Now;
-            DateTime endDate = data?.endDateTime ?? DateTime.Now.AddHours(12);
-
-            string meetingName = data?.subject ?? config.GetConnectionStringOrSetting(DefaultMeetingNameCfg);
+            // use defaults if required
+            requestModel.StartDateTime ??= DateTime.Now;
+            requestModel.EndDateTime ??= DateTime.Now.AddHours(1);
+            requestModel.Subject ??= config.GetConnectionStringOrSetting(ConfigConstants.DefaultMeetingNameCfg);
 
             try
             {
-                var authProvider = GetAuthenticationProvider(config);
+                var service = new BookingService(config);
+
+                var onlineMeeting = await service.CreateTeamsMeetingAsync(requestModel).ConfigureAwait(false);
 
-                var graphServiceClient = new GraphServiceClient(authProvider);
+                var newEvent = await service.CreateCalendarEventAsync(requestModel, onlineMeeting.JoinWebUrl).ConfigureAwait(false);
```

Comment:
The Event should be created only if explicitly requested via request input. Add a bool parameter to the input body such as "createEvent", defaults to false

Reply:
I Agree